Source/SwiftLintFramework/Rules/ModifierOrderRule.swift

```diff
@@ -0,0 +1,251 @@
+            "}",
+            "protocol Foo: class {} \n"                             +
+                "class Bar { \n"                                    +
+                "    private(set) public weak var foo: Foo? \n"     +
+            "} \n",
+            "open class Bar { \n"                                   +
+            "   open class func foo() -> Int { \n"                  +
+            "       return 42 \n"                                   +
+            "   } \n"                                               +
+            "} \n"                                                  +
+            "class Foo: Bar { \n"                                   +
+            "   class open override func foo() -> Int { \n"         +
+            "       return 43 \n"                                   +
+            "   } \n"                                               +
+            "}",
+            "open class Bar { \n"                                   +
+            "   open class func foo() -> Int { \n"                  +
+            "       return 42 \n"                                   +
+            "   } \n"                                               +
+            "} \n"                                                  +
+            "class Foo: Bar { \n"                                   +
+            "   open override class func foo() -> Int { \n"         +
+            "       return 43 \n"                                   +
+            "   } \n"                                               +
+            "}",
+            "@objc \n"                                              +
+            "final public class Foo: NSObject {}",
+            "@objcMembers \n"                                       +
+            "final public class Foo: NSObject {}",
+            "@objc \n"                                              +
+            "final open class Foo: NSObject { \n"                   +
+            "   weak open var weakBar: NSString? = nil \n"          +
+            "}",
+            "final public class Foo {} \n",
+            "internal class Foo: Bar { \n"                          +
+            "   internal override func bar() {} \n"                 +
+            "}",
+            "public struct Foo { \n"                                +
+            "   weak internal var weakBar: NSObjetc? = nil \n"      +
+            "}",
+            "class Foo { \n"                                        +
+            "   lazy internal var bar: String = \"foo\" \n"         +
+            "}"
+        ]
+    )
+
+    public func validate(file: File,
+                         kind: SwiftDeclarationKind,
+                         dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+
+        guard let offset = dictionary.offset else {
+            return []
+        }
+
+        let preferedOrderOfModifiers = [.atPrefixed] + configuration.preferedModifierOrder
```

Comment:
Why isn't `atPrefixed` simply part of the `preferedModifierOrder` configuration? Seems odd to do this unconditionally.

Reply:
All the attributes / modifiers starting with `@` are enforced by the compiler to be placed first. If the `atPrefixed` group would be in any other place than the first, it would mean all declarations using any modifier with the `@` prefix would fail.